packages/web-components/src/components/ic-tab-context/ic-tab-context.spec.ts

```diff
@@ -0,0 +1,369 @@
+import { newSpecPage } from "@stencil/core/testing";
+import { TabGroup } from "../ic-tab-group/ic-tab-group";
+import { TabPanel } from "../ic-tab-panel/ic-tab-panel";
+import { Tab } from "../ic-tab/ic-tab";
+import { TabContext } from "./ic-tab-context";
+
+describe("ic-tab-context component", () => {
+  it("should render tab context with tabs", async () => {
+    const page = await newSpecPage({
+      components: [TabContext, TabGroup, Tab, TabPanel],
+      html: `<ic-tab-context>
+      <ic-tab-group label="Example tab group">
+        <ic-tab>One</ic-tab>
+        <ic-tab>Two</ic-tab>
+        <ic-tab>Three</ic-tab>
+      </ic-tab-group>
+      <ic-tab-panel>Tab One</ic-tab-panel>
+      <ic-tab-panel>Tab Two</ic-tab-panel>
+      <ic-tab-panel>Tab Three</ic-tab-panel>
+      </ic-tab-context>`,
+    });
+
+    expect(page.root).toMatchSnapshot();
+
+    //test disconnected callback for automatic activation
+    page.setContent("");
+  });
+
+  it("should render tab context with tabs - light appearance", async () => {
+    const page = await newSpecPage({
+      components: [TabContext, TabGroup, Tab, TabPanel],
+      html: `<ic-tab-context appearance="light">
+      <ic-tab-group label="Example tab group" appearance="light">
+        <ic-tab>One</ic-tab>
+        <ic-tab>Two</ic-tab>
+        <ic-tab>Three</ic-tab>
+      </ic-tab-group>
+      <ic-tab-panel>Tab One</ic-tab-panel>
+      <ic-tab-panel>Tab Two</ic-tab-panel>
+      <ic-tab-panel>Tab Three</ic-tab-panel>
+      </ic-tab-context>`,
+    });
+
+    expect(page.root).toMatchSnapshot();
+  });
+
+  it("should render tab context with tabs - selected tab", async () => {
+    const page = await newSpecPage({
+      components: [TabContext, TabGroup, Tab, TabPanel],
+      html: `<ic-tab-context selected-tab-index="2">
+      <ic-tab-group label="Example tab group">
+        <ic-tab>One</ic-tab>
+        <ic-tab>Two</ic-tab>
+        <ic-tab>Three</ic-tab>
+      </ic-tab-group>
+      <ic-tab-panel>Tab One</ic-tab-panel>
+      <ic-tab-panel>Tab Two</ic-tab-panel>
+      <ic-tab-panel>Tab Three</ic-tab-panel>
+      </ic-tab-context>`,
+    });
+
+    expect(page.root).toMatchSnapshot();
+    expect(page.rootInstance.selectedTab).toBe(2);
+
+    const tabGroup = page.root.querySelector("ic-tab-group");
+
+    tabGroup.dispatchEvent(
```

Comment:
Just wondering what this is testing if the keydown handler is tested further down - also a bit confused why it's passing on line 76 because surely the selectedTab should be the first one when the Home key is pressed?

Reply:
> Just wondering what this is testing if the keydown handler is tested further down - also a bit confused why it's passing on line 76 because surely the selectedTab should be the first one when the Home key is pressed?

Have moved into a separate test. Intention was to cover the case where the selectedTabIndex is controlled, so the value does not change internally